app/main.py

```diff
@@ -1,14 +1,70 @@
-class Validator:
-    pass
+from abc import ABC, abstractmethod
+from typing import Generic, Type, TypeVar
 
+Instance = TypeVar("Instance")
```

Comment:
Do you need this here..?

Reply:
Can I keep proper generic class usage by creating class attributes like that?
```
buns = Number["BurgerRecipe", int](2, 3)
sauce = OneOf["BurgerRecipe", str](("ketchup", "mayo", "burger"))
```